src/Microsoft.NetFramework.Analyzers/Core/Helpers/SecurityDiagnosticHelpers.cs

```diff
@@ -274,7 +277,7 @@ public static string GetNonEmptyParentName(SyntaxNode current, SemanticModel mod
         /// For .NET Framework 4.X, this method returns the actual framework version instead of assembly verison of mscorlib,
         /// i.e. for .NET framework 4.5.2, this method return 4.5.2 instead of 4.0.0.0.
         /// </remarks>
-        public static Version GetDotNetFrameworkVersion(Compilation compilation)
+        public static Version? GetDotNetFrameworkVersion(Compilation compilation)
         {
             if (compilation == null || !IsTypeDeclaredInExpectedAssembly(compilation, "System.String", "mscorlib").GetValueOrDefault())
```

Comment:
@sharwell Do you know why compiler does not generate any warning for parameter not being declared nullable when the code has a null check up front?